### ISSUE TITLE
False positive: Ramllint should allow camelCase url parameters.

### ISSUE THREAD

**Issue by vanDonselaar** [QuickenLoans/ramllint#36]
Ramllint produces a false positive for the example from: http://raml.org/docs-200.html#parameters because it does not allow camelCase notation for url parameters.
In order to fix this, I think the regex [here](https://github.com/QuickenLoans/ramllint/blob/ac760bb8e4f66b2a85114516d3999f5f68748b13/src/defaults.json#L48) should be changed from:
```
^\\/([a-z]+(_[a-z]+)*|{[a-z]+(_[a-z]+)*})$
```
to
```
^\\/([a-z]+(_[a-z]+)*|{[a-z]+(_[a-zA-Z]+)*})$
                                   ^^^ 
```

Unrelated: I think you can improve the performance of the regexes by making some groups non-greedy like this:
```
^\\/([a-z]+?(_[a-z]+)*|{[a-z]+?(_[a-zA-Z]+)*})$
           ^                  ^
```

**Comment by kalisjoshua** [comment]
Currently the default rule follow what Quicken Loans' standards are. We don't allow for camelCase in our URLs.

The performance tip is quite helpful though. Thank you for that. I will keep this issue open for that alone.

**Comment by vanDonselaar** [comment]
> Currently the default rule follow what Quicken Loans' standards are. We don't allow for camelCase in our URLs.

Just to avoid confusion: I'm talking URL *parameters* here, not the actual URLs. The proposed regex still doesn't allow camelCased URL's as far as I know. The parameter name is only used in the RAML spec and is never used as part of a URL.

> The performance tip is quite helpful though. Thank you for that. I will keep this issue open for that alone.

NP :+1:

**Comment by kalisjoshua** [comment]
That is a good point. Thank you for the clarity. I will revisit this once my major refactor is complete.
